src/compiler/checker.ts

```diff
@@ -9257,6 +9326,11 @@ namespace ts {
                         if (!bodyDeclaration) {
                             bodyDeclaration = node;
                         }
+
+                        // abstract functions cannot have an implementation
+                        if (currentNodeFlags & NodeFlags.Abstract) {

```

Comment:
check that this is a method to avoid reporting multiple errors, e.g. abstract not allowed on a constructor, abstract methods can not have implementation.